packages/gensx-vercel-ai-sdk/src/index.tsx

```diff
@@ -9,30 +11,60 @@ export function createGSXComponent<TFn extends (...args: any[]) => any>(
   return gsx.Component<Parameters<TFn>[0], Awaited<ReturnType<TFn>>>(name, fn);
 }
 
-export const StreamObject: GsxComponent<
-  Parameters<typeof ai.streamObject>[0],
+// Define a more specific type for StreamObject that allows schema
```

Comment:
I added type tests and discovered that the types here were not quite right, so fixed them as well.